Chunks/SphericalChunkFactory.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Chunks.MarchingCubes;
+using Chunks.Voxels;
+using Common.Meshes;
+using OpenTK.Mathematics;
+
+namespace Chunks;
+public class SphericalChunkFactory
+{
+    private readonly ScalarFieldGenerator _scalarFieldGenerator;
+
+    private readonly Vector3i[] _sphereCenters;
+
+    private readonly float _globalScale;
+
+    public SphericalChunkFactory(ScalarFieldGenerator scalarFieldGenerator, Vector3i[] sphereCenters, float globalScale)
+    {
+        _scalarFieldGenerator = scalarFieldGenerator;
+        _sphereCenters = sphereCenters;
+        _globalScale = globalScale;
+    }
+
+    public List<Chunk> CreateSpheres(int chunksPerSide, bool generateVao = true)
+    {
+        if (chunksPerSide % 2 != 0)
+            throw new ArgumentException("# of chunks/side must be even");
+
+        List<Chunk> spheres = new();
+
+        for (int chunkX = -chunksPerSide / 2; chunkX < chunksPerSide / 2; chunkX++)
+        {
+            for (int chunkY = -chunksPerSide / 2; chunkY < chunksPerSide / 2; chunkY++)
+            {
+                Voxel[,,] averageScalarField0 = new Voxel[Chunk.Size, Chunk.Size, Chunk.Size];
```

Comment:
In euclidean we have Voxel[Chunk.Size +1, ...] better keep it consistant. Also I think it would be better to have a spharical scalar field generator so that logic in both is the same

Reply:
It got me wondering why we have `Chunk.Size + 1` elsewhere...
As of to the second suggestion, my reasoning was that `SphericalChunkFactory` is conceptually very close to `ChunkFactory`, i.e. they are both used in the respective chunk workers. `ScalarFieldGenerator` on the other hand is responsible for creating scalar field for the terrain & interacts directly with perlin noise stuff (`SphericalChunkFactory` doesn't)

Reply:
we have the +1 becuase this way the Chunk is Chunk.Size blocks long